### PR TITLE
Allow render to accept generics to type: Props, Events and Slots

### PR DESCRIPTION
**Pull request by ivoreis**
Allow the render method to accept generics to type: Props, Events and Slots.

Added extended options as per https://svelte.dev/docs#Creating_a_component
- target
- anchor
- props 
- context
- hydrate
- intro

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -3,11 +3,18 @@
...
-import { SvelteComponent } from 'svelte/types/runtime'
+import { SvelteComponentTyped } from 'svelte/types/runtime'
 
 export * from '@testing-library/dom'
 
-type SvelteComponentOptions = any
+export interface SvelteComponentOptions<P extends Record<string, any> = any>  {
+  target?: HTMLElement
+  anchor?: string
+  props?: P
```
Comment (andrew-secret) [inline types/index.d.ts:13]: awesome 🥰 this should work

**Comment by cbbfcd** [comment]
👍

**Comment by koenvg** [comment]
Hey, 

Can you add `context` and `target`  to the list of svelteComponentOptions so it reflects the typing definition?

Thanks

**Comment by ivoreis** [comment]
Sure, I'll update the PR. For reference https://github.com/sveltejs/svelte/blob/master/src/runtime/internal/dev.ts#L107-L115

**Comment by ivoreis** [comment]
I've extended `svelteComponentOptions` to also allow `context`. `Target`, on the other hand, will never be passed because is removed from `options` (https://github.com/testing-library/svelte-testing-library/pull/137/files#diff-2ef28f1bd92d5dcd1f2a04d56814d3adaee10cc939b4a7d7c861af3a3cbbccb7R11)

**Comment by yanick** [comment]
I rebased and merged to main manually. Thanks!

**Pull request closed without merging**